src/meta/processors/admin/SnapShot.cpp

```diff
@@ -77,78 +81,37 @@ bool Snapshot::getAllSpaces(std::vector<GraphSpaceID>& spaces, kvstore::ResultCo
 }
 
 cpp2::ErrorCode Snapshot::blockingWrites(storage::cpp2::EngineSignType sign) {
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
+    // check the index rebuild. not allowed to block write when index rebuilding.
+    auto prefix = MetaServiceUtils::rebuildIndexStatusPrefix();
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kv_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return cpp2::ErrorCode::E_BLOCK_WRITE_FAILURE;
+    }
+    while (iter->valid()) {
```

Comment:
why use while here?

Reply:
> why use while here?

I need check whether any index is being rebuild for all spaces and all indexes.